### PR TITLE
BUGFIX: Custom listenerClient API

### PR DESCRIPTION
**Pull request by mattv8**
This PR seeks to fix a bug I found with the Custom listenerClient API. Addresses issue #305

### CODE REVIEW

**Comment by JTF4** [comment]
I think this looks good. However, I'm going to do some digging to see why the JSON object is not parsed at the beginning to make sure we don't have a bigger issue.

File: src/index.ts [hunk 1]
```diff
@@ -331,10 +331,11 @@ function initialSetup() {
 			canBeFlashed (bool)
 			supportsChat (bool)
 			*/
-			obj = JSON.parse(String(obj)); //Re-parse JSON
 
 			logger(`Received JSON object: ${obj}`); //To make it easier for devs to verify the JSON objects they're sending
 
+			obj = JSON.parse(String(obj)); //Re-parse JSON
```
Comment (JTF4) [inline src/index.ts:337]: I would change this to 

`if(typeof obj !== 'object' && obj !== null) {
     logger(`Received JSON object: ${obj}`, 'info-quiet');
     obj = JSON.parse(String(obj)); //Re-parse JSON
}`

That way it does not parse an already parsed JSON object

**Review by JTF4 — CHANGES_REQUESTED** [review]

**Comment by JTF4** [comment]
@mattv8 if you can get this finished up today, I can get it included in the new beta release later today

**Comment by mattv8** [comment]
Good to go @JTF4. I wanted to take the time to verify with my software and listener clients first. Great suggestion to add a check before reparsing.

Be aware, this may also be an issue for the `listener_reassign_object` function as well, however we have the `listener_reassign` which doesn't take a JSON object-- that's been working well for my purposes.

**Review by JTF4 — APPROVED** [review]
Looks great! I'll test the other two functions you're referring to and see what they do

**Pull request merged**